### ISSUE TITLE
多个文档引会导致0名中

### ISSUE THREAD

**Issue by iCodyYang** [langgenius/dify#690]
## Dify version:
Cloud & Self Host 0.3.12

## 问题描述
「**云端版**」我在一个数据集下放了2个文档A与B。
A与B的内容相差很大，且文档格式清晰，采用「自动分段与清洗」与「高质量」引索，查看过处理后的分段没有问题。
```
# 1/ GM
GM的意思是“Good Morning”

# 2/ GN
GN的意思是“Good Night”
```

昨晚在chat中提问和数据集做命中测试，任何关于数据集中的内容都可以准确命中。无论来自A还是B

但是今天再测试，在没有任何改动的前提下，却只能命中文档A，提问B文档相关的问题也只能命中A。

之前使用「**社区版**」的时候（使用 Docker 部署在 Mac mini M1）也遇到过同样的问题，甚至2个文档都出现0命中。

## 尝试过，但无效的方案
- 「**社区版**」重装
- 「**社区版**」重新传文档
- 「**社区版**」将2个文档合并成一个文档使用

## 尝试过，但有效的方案
- 「**云端版**」将2个文档合并成一个文档使用
- 「**云端版**」重新传文档（但不确定后面是否还会出现问题）

**Comment by takatost** [comment]
This could be because the description of the dataset does not summarize the content of the dataset. The principle of selecting multiple datasets is to compare the problem with the description of the datasets and choose the most suitable dataset for retrieval.

**Comment by JohnJyong** [comment]
Have you tried re-uploading the document to confirm whether it is a program issue or data source issue?

**Comment by iCodyYang** [comment]
> Have you tried re-uploading the document to confirm whether it is a program issue or data source issue?

Yes, I have tried many times and I had my friends test it too, and they encountered the same problem

**Comment by iCodyYang** [comment]
> This could be because the description of the dataset does not summarize the content of the dataset. The principle of selecting multiple datasets is to compare the problem with the description of the datasets and choose the most suitable dataset for retrieval.

抱歉，为了更精准的表达，我就用中文回复了
1. 我是在1个数据集内上传的2个文档，并非多个数据集
2. 我尝试过按照文档给数据集添加描述，但是效果更差了。我认为只有一个数据集，并且我在前置提示词中写明只能依据数据库回答情况下，不应该不知道选择哪个数据库。
3. 我遇到的问题是，之前一切正常，但是过了一天后，没有任何改动的情况下，开始出现0命中。
thx

**Comment by KomanLiang** [comment]
我也出现了这个问题 上下文的配置感觉没有生效 文档都是0命中 也没有根据文档切片进行参考回答
